### DISCUSSION TITLE
storing python class instance in a struct

### DISCUSSION THREAD

**Discussion by JasonGrace2282** [PyO3/pyo3#4246] [Questions] [answered]
If I have a class like
```py
class PythonClass:
    def some_method(self):
        ...
```
Is there any way I can store an instance of this class in a `pyclass`? Something like
```rust
pub trait PythonClassTrait {
  fn some_method(self);
}

#[pyclass]
struct RustStruct {
  attr: Py<dyn PythonClassTrait>
}
```
?

**Comment by davidhewitt** [comment, marked as answer]
Not as a `dyn Trait`. At the moment you should store `Py<PyAny>` and go through functions like `.call_method("some_method", args, kwargs)`.

**Comment by JasonGrace2282** [comment]
Ah, interesting, thanks a lot :heart:. Are there any plans to add type checking via traits (or is it even possible to do so)?

Also, do you think it's worth mentioning this in the docs?

Speaking of docs, I forgot to say this earlier, but thanks so much for your work! PyO3 is extremely well documented and easy to use :)

**Reply by davidhewitt** [reply]
I think using traits _might_ be possible, but I haven't explored it technically at all and I think it's a long way off. There's a lot of work to complete first on things like free threading support. 

I'm glad the documentation has been a success for you! We try hard, though I always think it could be better 😬

**Reply by davidhewitt** [reply]
> Also, do you think it's worth mentioning this in the docs?

I think there's definitely room for some topics in the docs on this kind of interaction. It's quite common (and related to e.g. subclasses in python).
